### PR TITLE
Fix fully CNN model attention

### PR DESCRIPTION
**Pull request by maksymbevza**
The fully convolutional model has used attention in a wrong way.
The shape of `pre_attn` is `[bs, dec_seq, enc_seq]`
And according to https://github.com/pytorch/pytorch/issues/1020 the softmax is done over dimension 0, whereas it should be done over dimension 2.

### CODE REVIEW

**Comment by srush** [comment]
Thanks. Have you been able to get this model to work? It was a user contribution, so we haven't tested it much yet.

**Comment by maksymbevza** [comment]
With this bugfix, the model becomes sane at inference time. And obtains pretty good scores, by the way. However, the scores with OpenNMT-py fully CNN model were lower compared to my experiments with FAIR Seq implementation.

**Comment by srush** [comment]
Okay, let us know if you have exact numbers.

**Comment by patelrajnath** [comment]
@maksymbevza
What configuration did you use? I'm trying to train a model with the following configuration to compare with fairseq-py default configuration. 

python -m opennmt.train \
        -gpuid 1 \
        -encoder_type cnn \
        -decoder_type cnn \
        -enc_layers 4 \
        -dec_layers 3 \
        -rnn_size 256 \
        -src_word_vec_size 256 \
        -tgt_word_vec_size 256 \
        -epochs 30 \
        -data data-bin_opennmt/en-de \
        -save_model cnn-model

The scores are quite low (14 BLEU ) of this model in comparison to fairseq-py (30 BLEU). BTW I am using the IWSLT14 data set given as a an example with fiarseq-py.

**Comment by hcharaya** [comment]
Did you able to get the accepted BLEU score using CNN model i tried training using 
python train.py -data iwslt14.tokenized.de-en/geToEn -save_model cnn-model_13_nov -encoder_type cnn -decoder_type cnn -world_size 1 -gpu_ranks 0 -report_every 4 -batch_size 16 -dropout 0.2 -learning_rate 0.25 -max_generator_batches 16 -valid_batch_size 16 -train_steps 2000000 -enc_layers 4 -dec_layers 3 -src_word_vec_size 256 -tgt_word_vec_size 256 -rnn_size 256

and then inference using 
python translate.py -model cnn-model_13_nov_step_255000.pt -src iwslt14.tokenized.de-en/test.de -output pred_255000.txt -replace_unk -verbose -gpu 1 -batch_size 16

however i am getting very low BLEU score (~2).
Please let me know if i missed anything and in prediction i am getting several words are repeated

**Comment by vince62s** [comment]
yes there is a bug, I'll post a fix.
if you intend to tune with it, please post here: #693
